libs/cart/guides/extension.md

```diff
@@ -0,0 +1,106 @@
+# Extension
+
+`@daffodil/cart` provides a number of extension mechanisms so that it can be customized to fit specific needs.
+
+## Generic Models
+
+All Daffodil layers can operate on generic extensions of vanilla Daffodil models. Custom models can therefore be used while retaining type safety. The following example illustrates customizing the cart model with the cart facade.
+
+```typescript
+import {
+  DaffCartFacade,
+  DaffCart
+} from '@daffodil/cart';
+
+interface CustomCart extends DaffCart {
+  customField: string;
+}
+
+@Component({})
+class CartComponent implements OnInit {
+  cart$: Observable<CustomCart>;
+
+  constructor(private cartFacade: DaffCartFacade<CustomCart>) {}
+
+  ngOnInit() {
+    // this.cartFacade.cart$ is of type CustomCart
+    this.cart$ = this.cartFacade.cart$;
+  }
+}
+```
+
+## Extensible GraphQL Fragments
+
+Arbitrary additional fields can be requested on the cart object. Inject a GraphQL document node containing fragments on the platform's cart type to define extra fields.
+
+The following cart drivers support extensible fragments:
+- Magento
```

Comment:
I think we don't need to say what the supported platforms here. I think for all new drivers, we'll just have them all support extensible fragments. So as long as we say somewhere that magento is the only platform supported by cart, we should be good.

Reply:
Well only the drivers that use GraphQL support this. So its not guaranteed that all future drivers will support this. Also, the In-Memory driver does not.

Reply:
I think we should absolutely explicitly state which drivers support this mechanism of extension.

Reply:
Ok, maybe we should also say: "Only drivers that use GraphQL support extensible fragments because fragments are specific to GraphQL."

Reply:
added